libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/IDPInititatedLoginReceiver.java

```diff
@@ -57,18 +62,41 @@ public void onReceive(Context context, Intent intent) {
                 final Bundle extras = intent.getExtras();
                 if (extras != null) {
                     userHint = extras.getString(USER_HINT_KEY);
+                    spActivityName = extras.getString(SP_ACTVITY_NAME_KEY);
                 }
 
                 // Launches login flow if the user doesn't already exist on the SP app.
                 if (!doesUserExist()) {
                     launchLoginActivity();
+                } else {
+
+                    // If user hint was passed in, switches to the specified user.
+                    UserAccountManager.getInstance().switchToUser(getUserForHint());
+
+                    /*
+                     * If the IDP app specified a component to launch after login, launches that
+                     * component. The activity that is passed in must contain its full package name.
+                     */
+                    if (!TextUtils.isEmpty(spActivityName)) {
+                        try {
+                            final Intent launchIntent = new Intent(SalesforceSDKManager.getInstance().getAppContext(),
+                                    Class.forName(spActivityName));
+                            launchIntent.addCategory(Intent.CATEGORY_DEFAULT);
+                            SalesforceSDKManager.getInstance().getAppContext().startActivity(launchIntent);
+                        } catch (Exception e) {
```

Comment:
We should catch `Exception` here and log it to prevent a crash on the SP app if a bogus component is passed in.